### ISSUE TITLE
Cannot import cv2 after conda install

### ISSUE THREAD

**Issue by aspen1135** [conda-forge/opencv-feedstock#170]
Can't import module cv2 after installing latest opencv package from condaforge (4.1.0).

OS:
Windows 7 Ultimate x64

```
$ conda install -c conda-forge opencv
```

```
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.
```

`import cv2` Note: PyCharm IDE detects "No Module named cv2"

Run:
```
INTEL MKL ERROR: The operating system cannot run %1. mkl_intel_thread.dll.
Intel MKL FATAL ERROR: Cannot load mkl_intel_thread.dll.
```

I have also tried downloading the packages from the main OpenCV website and setting the environment manually. This was what was discussed on stackoverflow [here](https://stackoverflow.com/questions/23119413/how-do-i-install-python-opencv-through-conda) and also instructed through an installation guide [here](https://www.youtube.com/watch?v=EcFtefHEEII) 

I extracted `opencv` folder from the downloaded executable package [here](https://sourceforge.net/projects/opencvlibrary/)  to `C:\OpenCV 4.1.0` then set a new system variable named `OPENCV_DIR` with the path `C:\OpenCV 4.1.0\opencv\build\x64\vc15`.

Then added `%OPENCV_DIR%\bin.` to PATH environment in system variables.

Neither installation methods work. It seems these guides may be outdated for new 4.1 versions of OpenCV. What am I missing?

And yes- IDE interpreter has been configured to use the installed Python 3 executable and corresponding anaconda environment. Installation of other modules work fine using conda install.

**Comment by hmaarrfk** [comment]
You should not have to download opencv yourself when using anaconda. 

Have you found 
https://github.com/conda-forge/opencv-feedstock/issues/165
Maybe they can help you better understand what is going on?

**Comment by aspen1135** [comment]
> You should not have to download opencv yourself when using anaconda.
> 
> Have you found
> #165
> Maybe they can help you better understand what is going on?

Thats why I raised an issue. I have the package installed in anaconda but I cannot import it. The thread you mentioned is almost a month old and was marked as closed, I do not think that is the right place to raise the issue.

**Comment by hmaarrfk** [comment]
It was marked as closed because it turned out to be a badly configure install.

My comment there about not using conda forge unless you mean it still holds

I wouldn't use the standard prompt I'm windows.
I don't think I have any other environment variables in my windows install.

Maybe try to reinstall anaconda. When you are installing opencv, do not specify conda forge. It should find the best combination of opencv and dependencies for your system

**Comment by aspen1135** [comment]
> Maybe try to reinstall anaconda. When you are installing opencv, do not specify conda forge. It should find the best combination of opencv and dependencies for your system

When I was checking the repositories, the conda-forge distribution was the most up to date. If I'm installing a module, wouldn't I want the latest version? The others are running version 3 where conda-forge has version 4 uploaded.

+ I also tried installing from menpo without success. I am re-installing anaconda now to see if it resolves any issues. I will update this comment after I do.

**Comment by hmaarrfk** [comment]
I understand your confusion, but please take some time to read the discussion in https://github.com/conda-forge/opencv-feedstock/issues/165
where many of the Anaconda employees, and those involved in conda forge have already chimed in.

Generally, you will want either all conda-forge, or all defaults.

**Comment by hmaarrfk** [comment]
https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge

https://conda-forge.org/docs/user/tipsandtricks.html

**Comment by aspen1135** [comment]
I have reinstalled anaconda completely now with all other dependencies needed for my project again but the problem still persists. It's also worth noting that cv2 isn't listed as a package in my IDE (pycharm), but conda detects 'opencv' being installed when trying to uninstall or reinstall it.

I have followed your other thread closely. I tried your first solution by using the defaults installation:
```
conda uninstall opencv
conda install -c defaults opencv
```

installation runs successfully. However, cv2 still fails to import, which prints:
```
INTEL MKL ERROR: The operating system cannot run %1. mkl_intel_thread.dll.
Intel MKL FATAL ERROR: Cannot load mkl_intel_thread.dll.
```

According to the [link](https://conda.io/projects/conda/en/latest/user-guide/troubleshooting.html#numpy-mkl-library-load-failed) there are a few other things to try such as:

Removing any MKL files from system32. There were a few that exist on my computer such as `mkl_core.dll mkl_def.dll mkl_intel_thread.dll mkl_sequential.dll`. Removing these to a seperate folder my on my user desktop did not change anything.

I tried activating the environment with `conda update -c defaults python` and the problem still persists as well.

I also tried changing the search-modification option with `set CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1`. Problem still occurs.

None of the other errors on that link's list correspond to what I am experiencing. Here is what my PATH environment looks like right now:

`C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\ProgramData\chocolatey\bin;C:\ProgramData\Anaconda3\Library\bin;`

Note: `C:\ProgramData\Anaconda3\Library\bin;` environment is needed for my PyAV (av) module to work. Removing this from PATH does not seem to have any effect on the problem.

**Comment by aspen1135** [comment]
Interestingly, I tried installing this on my laptop. My IDE still cannot locate the cv2 module, but the modules functions such as img.read() seems to work just fine.

I do not do my development work on my laptop very often though and I'd like to find the solution to this issue still.

**Comment by hmaarrfk** [comment]
what is the output of `conda list` from your command prompt

**Comment by hmaarrfk** [comment]
I'm not sure if the world is conspiring against you, but maybe the easiest thing for you to do is to jump all in with conda-forge.

Are you using a seperate environment for your python packages that you install? That is generally the best thing. Best to leave the `base` or `root` environment just for `conda`

https://github.com/mikeboers/PyAV#installation

I'm not really a big fan of people saying:
```
conda install -c conda-forge opencv
```

but 
```
conda config --add channels conda-forge
conda create --name opencv
conda activate opencv
conda install opencv av ipython spyder jupyter
```

is probably the best way to go for you.

Have you tried adding the conda-forge channel all out before?

**Comment by aspen1135** [comment]
No, but as long as it gets the job done I don't think I care too much.

If I change my primary repository to conda-forge like this, does that mean I should not be using standard pip-installs either anymore?

I'm kind of at this turning point in my project where I'm not sure if python is going to be the best language to continue working in anyways. Right now I rely very heavily on FFmpeg for my backend and the available python modules that support it either have A) performance issues or B) limited API support. I wanted to try and get video encoding up and running (my program is primarily audio-oriented at the moment), but I need the opencv library to do so.

Basically, I just wanted to use opencv for a brief moment to do some bench-marking.

**Comment by hmaarrfk** [comment]
Generally you don't want to use pip and conda at the same time. You run into a few missing packages l in defaults, less so in conda-forge.

When I do install packages with pip, I do it with a float that disabled dependencies then install the dependencies manually. Otherwise you run into incompatible packages requiring exclusive dlls like you are now.

**Comment by mshabunin** [comment]
Check this issue: https://github.com/ContinuumIO/anaconda-issues/issues/10213
Do you have Intel Compiler or Intel MKL installed on your system? It seem to be incompatible with MKL from conda-forge.

**Comment by hmaarrfk** [comment]
Is this still an issue?

**Comment by suryacaprice** [comment]
Yes i have the same issue

**Comment by hmaarrfk** [comment]
Can you please open a new issue showing us your full environment

**Comment by ankitshah009** [comment]
Temporary fix - Most recent version of Conda has some problems with opencv installation. 

> conda config --set allow_conda_downgrades true
> conda install conda=4.6.14
> conda install -c conda-forge opencv

**Comment by hmaarrfk** [comment]
The original issue was reported in June 2019. It is very likely that you are experiencing a different issue.

Please open a new request so we can best help you.

**Comment by patricksnape** [comment]
Closing old issue as it's unlikely this issue is persisting in exactly the form originally reported. Anyone still having issues please open a new requst

**Comment by aujinen** [comment]
Environment: Windows10, anaconda, python3.7.6

First time:
conda install -c conda-forge opencv
I got "opencv-4.1.1"
---> error occured (as "No module of CV2")

Second time:
conda uninstall opencv
conda install -c conda-forge opencv=3
opencv3.4.2
No-error, OK!

I think lower version of opencv is suitable for python3.* system.

**Comment by alelom** [comment]
Issue still here. And it's damn frustrating.

**Comment by hmaarrfk** [comment]
@alelom the best way for us to help you is if you open a new issue.

We try to respond pretty quickly

**Comment by patricksnape** [comment]
Also note that I just re-read the issue and that installing any conda-forge packages inside a defaults installation is generally not supported and unlikely to work. Please create a new environment containing only conda-forge packages and then post an issue if the problem persists.
